cronjob_scripts/trigger-package-build.py

```diff
@@ -0,0 +1,217 @@
+    # Note that the old next-unbuilt-package.sh script actually checked the top 4 * CRATE_CHECK_LIMIT
+    # crates from get-stats.sh + popular-crates.txt rather than shuffling the whole lot before checking.
+    # The original script just looped through the whole list, but kept track of where it got to,
+    # to avoid getting stuck on unbuildable crates.
+    #
+    # For now I'm trying to be roughly backwards compatible with the old script's numbers,
+    # but I think that shuffling first is an important bugfixt. Once we are using crates.io sparse
+    # index api without rate-limits, we can think about checking more crates.
+    crates_to_check = random.sample(list(possible_crates), 4 * check_limit)
+    if os.environ.get("RECHECK"):
+        crates_to_check = ["cargo-quickinstall"] + crates_to_check
+
+    repo_url = get_repo_url()
+    branch = get_branch()
+    builds_scheduled = 0
+    for crate in crates_to_check:
+        print(f"Checking {crate} for {target_arch}")
+        version = get_current_version_if_unbuilt(
+            repo_url=repo_url, crate=crate, target_arch=target_arch
+        )
+        if not version:
+            continue
+
+        no_default_features = "false"
+        if crate == "gitoxide":
+            features = "max-pure"
+            no_default_features = "true"
+        elif crate == "sccache":
+            features = "vendored-openssl"
+        else:
+            features = ",".join(
+                [feat for feat in version["features"].keys() if "vendored" in feat]
+            )
+
+        workflow_run_input = {
+            "crate": crate,
+            "version": version["vers"],
+            "target_arch": target_arch,
+            "features": features,
+            "no_default_features": no_default_features,
+            "build_os": build_os,
+            "branch": branch,
+        }
+        print(f"Attempting to build {crate} {version['vers']} for {target_arch}")
+        subprocess.run(
+            ["gh", "workflow", "run", "build-package.yml", "--json", f"--ref={branch}"],
+            input=json.dumps(workflow_run_input).encode(),
+            check=True,
+        )
+        builds_scheduled += 1
+        if builds_scheduled >= check_limit:
+            print("Scheduled enough builds, exiting")
+            break
+        time.sleep(30)
+
+
+def get_build_os(target_arch: str) -> str:
+    try:
+        return TARGET_ARCH_TO_BUILD_OS[target_arch]
+    except KeyError:
+        raise ValueError(f"Unrecognised target arch: {target_arch}")
+
+
+def get_target_architectures() -> list[str]:
+    target_arch = os.environ.get("TARGET_ARCH", None)
+    if target_arch in TARGET_ARCH_TO_BUILD_OS:
+        return [target_arch]
+
+    if target_arch == "all":
+        return list(TARGET_ARCH_TO_BUILD_OS.keys())
+
+    try:
+        rustc_version_output = subprocess.run(
+            ["rustc", "--version", "--verbose"], capture_output=True, text=True
+        )
```

Comment:
```suggestion
    rustc_version_output = subprocess.run(
         
["rustc", "--version", "--verbose"], capture_output=True, text=True
    )

    try:
```

Since the `except` access `rustc_version_output`, it can't handle the error from `subprocess.run`

Reply:
cc @alsuren in case you missed this

Reply:
cheers. I had a bit of clean up of my blanked except: blocks just now, so hopefully it's a bit cleaner now.